gms/gossiper.cc

```diff
@@ -2311,6 +2311,9 @@ future<> gossiper::wait_alive(std::vector<gms::inet_address> nodes, std::chrono:
 }
 
 future<> gossiper::wait_for_live_nodes_to_show_up(size_t n) {
+    if (this_shard_id() != 0) {
```

Comment:
Why this change (and again, it's not really part of the main change)?
This function doesn't work on other shards? Why?

Reply:
`_pending_mark_alive_endpoints` is updated only on shard 0 hence the need.
That said, we can use invoke_on to run the body explicitly on shard 0, but since it is called only on shard 0 in practice, we can just assert that.